test/demo/raiseIntent/intentResolution.inttest.ts

```diff
@@ -0,0 +1,215 @@
+        });
+    }), 1],
+    ['a delayed async value', appHandlingIntent, jest.fn(async () => {
+        await delay(1000);
+        return 1;
+    }), 1],
+    ['a literal', appHandlingIntent, jest.fn(() => 1), 1]
+];
+
+// E.g. When an intent handler returns an error
+
+describe('Intent resolution', () => {
+    describe.each(resolutionValueTests)(
+        'When an intent handler returns %s',
+        (testTitle: string, intentHandlingApp: TestAppData, listener: ContextListener | undefined, expectedValue: any) => {
+            setupQuitAppAfterEach(intentHandlingApp);
+
+            beforeEach(async () => {
+                await openDirectoryApp(intentHandlingApp);
+            });
+
+            describe('When an app has 1 window', () => {
+                test('The expected value is resolved', async () => {
+                    await fdc3Remote.addIntentListener(intentHandlingApp, preregisteredIntent.type, listener);
+                    const resolution = raiseIntent(preregisteredIntent, intentHandlingApp);
+                    await testExpectedResolution(resolution, expectedValue);
+                });
+
+                test('And has multiple intent handlers, the expected value is resolved', async () => {
+                    const listeners = new Array(4).fill(null).map(async (_, i) => {
+                        return fdc3Remote.addIntentListener(intentHandlingApp, preregisteredIntent.type, createDelayedFunction(() => i, 2000 * (i + 1)));
+                    });
+                    await Promise.all(listeners);
+                    const resolution = raiseIntent(preregisteredIntent, intentHandlingApp);
+                    await testExpectedResolution(resolution, expectedValue);
+                });
+            });
+
+            describe('When an app has multiple windows', () => {
+                let children: Identity[];
+                const WINDOW_COUNT = 3;
+
+                beforeEach(async () => {
+                    children = await createChildWindows(intentHandlingApp, WINDOW_COUNT);
+                });
+
+                test('The expected value is resolved', async () => {
+                    await fdc3Remote.addIntentListener(intentHandlingApp, preregisteredIntent.type, listener);
+                    const resolution = raiseIntent(preregisteredIntent, intentHandlingApp);
+
+                    await testExpectedResolution(resolution, expectedValue);
+                });
+
+                test('And a child window has the only intent listener, the expected value gets resolved', async () => {
+                    await fdc3Remote.addIntentListener(children[1], preregisteredIntent.type, listener);
+                    const resolution = raiseIntent(preregisteredIntent, intentHandlingApp);
+
+                    await testExpectedResolution(resolution, expectedValue);
+                });
+
+                test('And all the windows return a value, the first to return is the value resolved', async () => {
```

Comment:
I was missing the intent listener on the `intentHandlingApp`. Fixed now.

Reply:
Ah, I see. The test would still pass if the third listener to return is the value we use 😛.

Suggest change to something like `createDelayedFunction(() => i, 2000 * (i + 1)));` to `createDelayedFunction(() => {not: expectedValue}, 2000 * (i + 1)));`

Reply:
Now its just a string so it can't pass. I do think its important to have a unique value that can be used to see what one is returning first.

Reply:
Cool. Think it would be clearer as `createDelayedFunction(() => `delayed-value-${i}`, 2000 * (i + 1)))`, and then it becomes obvious what `expect = `delayed-value-0` is about.

And would be tempted to do

    await expect(resolution).resolves.toHaveProperty('data', expectedValue !== undefined ? expectedValue : `delayed-value-0`);

instead of the `if` block and the new variable. Would also change "the first to return" to "the first to return a non-undefined value" in the test title